agent/taskresource/cgroup/cgroup.go

```diff
@@ -97,12 +109,95 @@ func (c *CgroupResource) GetName() string {
 	return resourceName
 }
 
+// DesiredTerminal returns true if the cgroup's desired status is REMOVED
+func (c *CgroupResource) DesiredTerminal() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.GetDesiredStatus() == taskresource.ResourceStatus(CgroupRemoved)
+}
+
+// KnownCreated returns true if the cgroup's known status is CREATED
+func (c *CgroupResource) KnownCreated() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.GetKnownStatus() == taskresource.ResourceStatus(CgroupCreated)
+}
+
+// TerminalStatus returns the last transition state of cgroup
+func (c *CgroupResource) TerminalStatus() taskresource.ResourceStatus {
+	return taskresource.ResourceStatus(CgroupRemoved)
+}
+
+// GetNextKnownStateProgression returns the state that the resource should
+// progress to based on its `KnownState`.
+func (c *CgroupResource) GetNextKnownStateProgression() taskresource.ResourceStatus {
+	return c.GetKnownStatus() + 1
+}
+
+// TransitionFunction calls the function required to move to the specified status
+func (c *CgroupResource) TransitionFunction(nextState taskresource.ResourceStatus) (bool, error) {
+	transitionFunc, ok := c.transitionFunctionMap()[nextState]
+	if !ok {
+		seelog.Criticalf("Cgroup Resource [%s]: unsupported desired state transition [%s]: %s",
+			c.taskARN, c.GetName(), nextState.String())
+		return false, nil
+	}
+	return true, transitionFunc()
+}
+
+func (c *CgroupResource) transitionFunctionMap() map[taskresource.ResourceStatus]func() error {
```

Comment:
Why you need this function to return the map, you can just use the `transitionFunctionMap` above?